ksp_plugin/identification.hpp

```diff
@@ -14,5 +23,28 @@ using GUID = std::string;
 // corresponds to |uint32_t|.
 using PartId = std::uint32_t;
 
+// Comparator by PartId.  Useful for ensuring a consistent ordering in sets of
+// pointers to Parts.
+struct PartByPartIdComparator {
+  bool operator()(not_null<Part*> left, not_null<Part*> right) const;
+  bool operator()(not_null<Part const*> left,
+                  not_null<Part const*> right) const;
+};
+
+// Comparator by GUID.  Useful for ensuring a consistent ordering in sets of
+// pointers to Vessels.
+struct VesselByGUIDComparator {
+  bool operator()(not_null<Vessel*> left, not_null<Vessel*> right) const;
+  bool operator()(not_null<Vessel const*> left,
+                  not_null<Vessel const*> right) const;
+};
+
```

Comment:
Could we name the types
```cpp
std::map<not_null<Part*>, T, PartByPartIdComparator>
std::set<not_null<Vessel*>, VesselByGUIDComparator>
std::set<not_null<Vessel const*>, VesselByGUIDComparator>
```
they're a bit of a mouthful, and it's not like we want to mix with comparatorless maps or sets of the same.